server/test/createJSONFromACHFiles_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/moov-io/ach"
+	"github.com/moov-io/ach/server"
+	"github.com/moov-io/base"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+// TestCreateJSONFromACHFiles creates JSON from existing ACH Files
+func TestCreateJSONFromACHFiles(t *testing.T) {
+	files := getFileNames()
+
+	for _, file := range files {
+		f, err := os.Open(file.AchFileName)
+		if err != nil {
+			log.Fatal(err)
+		}
+		r := ach.NewReader(f)
+		achFile, err := r.Read()
+		if err != nil {
+			fmt.Printf("Issue reading file: %+v \n", err)
+		}
+		// ensure we have a validated file structure
+		if achFile.Validate(); err != nil {
+			fmt.Printf("Could not validate entire read file: %v", err)
+		}
+		// If you trust the file but it's formatting is off building will probably resolve the malformed file.
+		if achFile.Create(); err != nil {
+			fmt.Printf("Could not build file with read properties: %v", err)
+		}
+
+		f.Close()
+
+		fmt.Printf("ACH File: %v \r\n", file.AchFileName)
+
+		// ENR ACH Files does not have BatchHeader.EffectiveEntryDate, so setting this to Today +1 to be included
+		// in the JSON File.  For this test after the ACH file is converted to JSON, the test validates the JSON by
+		// calling ach.FileFromJSON(bs) and it fails with an empty date time.
+		if file.SECCode == "ENR" {
+			for _, batch := range achFile.Batches {
+				batch.GetHeader().EffectiveEntryDate = base.NewTime(time.Now().AddDate(0, 0, 1))
+			}
+
+		}
+
+		bs, err := json.MarshalIndent(achFile, "", " ")
+		if err != nil {
+			fmt.Println("error:", err)
+		}
+
+		fmt.Printf("JSON Output: %v \r\n", string(bs))
+
+		fmt.Printf("Validating JSON byte stream %v \r\n", file.jsonName)
+
+		// Create ACH File from JSON
+		file, err := ach.FileFromJSON(bs)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		// Validate ACH File
+		if file.Validate(); err != nil {
+			fmt.Printf("Could not validate file: %v", err)
+		}
+
+		repository := server.NewRepositoryInMemory()
+		s := server.NewService(repository)
+
+		// Create and store ACH File in repository
+		fileID, err := s.CreateFile(&file.Header)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		// Get the stored ACH File from repository
+		getFile, err := s.GetFile(fileID)
+		if err == server.ErrNotFound {
+			t.Errorf("expected %s received %s w/ error %s", "ErrNotFound", getFile.ID, err)
+		}
+
+	}
+}
+
+type fileNames struct {
+	SECCode     string
+	AchFileName string
+	jsonName    string
+}
+
+func getFileNames() []fileNames {
```

Comment:
FYI, there's `path/filepath.Glob` that wouldn't require we list every file manually.

https://golang.org/pkg/path/filepath/#Glob

Reply:
I know, but I wanted to control that.  It seemed risky to just do a pattern match.  if you want  a pattern match what would you want it on?